googleapiclient/discovery.py

```diff
@@ -355,8 +355,9 @@ def build_from_document(
     scopes = list(
       service.get('auth', {}).get('oauth2', {}).get('scopes', {}).keys())
 
-    # If so, then the we need to setup authentication.
-    if scopes:
+    # If so, then the we need to setup authentication if no developerKey is
+    # specified.
+    if scopes and not developerKey:
```

Comment:
scopes can't be passed in, it's gather from the discovery doc. Scopes indicate that the service requires some sort of authentication, but if the user provides a developerKey then authentication is handled by that.